packages/router/utils.ts

```diff
@@ -198,7 +198,9 @@ type _PathParam<Path extends string> =
     ? _PathParam<L> | _PathParam<R>
     : // find params after `:`
     Path extends `:${infer Param}`
-    ? Param
+    ? Param extends `${infer Optional}?`
+      ? Optional
+      : Param
```

Comment:
Gotta sync up with @pcattori here - this makes sure we get the right params names without the `?` but it currently requires a key in the `Params` type